codbex-electra/opencart/listeners/sales-order-handler.js

```diff
@@ -0,0 +1,84 @@
+import { database } from "@dirigible/db";
+import { logging } from "@dirigible/log";
+
+const updateStatement = `
+	UPDATE oc_order 
+	SET
+		invoice_no = ?,
+		invoice_prefix = ?,
+		store_id = ?,
+		customer_id = ?,
+		comment = ?,
+		total = ?,
+		order_status_id = ?,
+		tracking = ?,
+		language_id = ?,
+		currency_id = ?,
+		accept_language = ?,
+		date_modified = ?
+	WHERE order_id = ?
+`;
+
+const logger = logging.getLogger("codbex-electra.opencart.listeners.sales-order-handler.js");
+
+exports.onMessage = function (messageString) {
+	logger.info("Processing sales order message [{}]", messageString);
+
+	const message = JSON.parse(messageString);
+	const operation = message.operation;
+
+	if (operation === 'update') {
+		handleUpdate(message.entity);
+	} else {
+		logger.debug("Message [{}] will not be handled.", messageString);
+	}
+};
+
+function handleUpdate(salesOrder) {
+	const connection = database.getConnection("OpenCartDB");
+	let statement;
+	let resultSet;
+	try {
+		statement = connection.prepareStatement(updateStatement);
+
+		statement.setInt(1, salesOrder.InvoiceNumber);
+		statement.setString(2, salesOrder.InvoicePrefix);
+		statement.setInt(3, salesOrder.Store);
+		statement.setInt(4, salesOrder.Customer);
+		statement.setString(5, salesOrder.Comment);
+		statement.setDouble(6, salesOrder.Total);
+		statement.setInt(7, salesOrder.Status);
+		statement.setString(8, salesOrder.Tracking);
+		statement.setInt(9, salesOrder.Language);
+		statement.setInt(10, salesOrder.Currency);
+		statement.setString(11, salesOrder.AcceptLanguage);
+
+		const Timestamp = Java.type('java.sql.Timestamp');
+		const dateModified = new Timestamp(Date.now());
+		statement.setTimestamp(12, dateModified);
+
+		const id = parseInt(salesOrder.Id);
+		statement.setInt(13, id);
+
+		const updatedRows = statement.executeUpdate();
+		logger.info("[{}] rows updated", updatedRows);
+	} catch (err) {
+		logger.error('Failed to update SalesOrder in OpenCart DB. \nError: [{}]', err.message);
+		throw err;
+	} finally {
+		closeResources(resultSet, statement, connection);
+	}
+}
+
+function closeResources(...resources) {
+	resources.forEach(r => {
+		if (r != undefined) {
```

Comment:
It's probably better to use:
```JS
if (r) {
  ...
}
```
That way you would also check for `null` and it's a bit shorter

Reply:
done